packages/workspace/src/browser/workspace-input-dialog.ts

```diff
@@ -54,4 +57,24 @@ export class WorkspaceInputDialog extends SingleTextInputDialog {
         // Add the path and icon div before the `inputField`.
         this.contentNode.insertBefore(element, this.inputField);
     }
+
+    /**
+     * Append the human-readable absolute root `path` to the dialog.
+     */
+    protected appendRootName(): void {
+        const root = this.workspaceService.getWorkspaceRootUri(this.props.parentUri);
+        if (root) {
+            // computes the label for the parent URI
+            const filename = this.labelProvider.getLongName(root);
```

Comment:
I'm thinking that we display the simple name of the root (ex: `theia`) and not the path.
If we happen to have two roots with the same exact name then we can fallback to displaying the full path like you've done. This will keep the root name simple in most cases.